operator/controllers/reconcilers/server/statefulset_reconciler_test.go

```diff
@@ -463,7 +466,13 @@ func TestToStatefulSet(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			statefulSet := toStatefulSet(test.meta, test.podSpec, test.volumeClaimTemplates, test.scaling, test.labels, test.annotations)
+			statefulSet := toStatefulSet(test.meta,
+				test.podSpec,
+				test.volumeClaimTemplates,
+				test.scaling,
+				test.statefulSetPersistentVolumeClaimRetentionPolicy,
```

Comment:
While you're referencing `test.statefulSetPersistentVolumeClaimRetentionPolicy` here, this remains null in all the test cases created above. I would add a test case for this.

Reply:
I have updated a current test, to have PersistentVolumeClaimRetentionPolicy in it :)